pkg/sidecar_mounter/sidecar_mounter.go

```diff
@@ -213,25 +214,34 @@ func logVolumeTotalSize(dirPath string) {
 }
 
 // collectMetrics collects metrics from the gcsfuse instance every 10 seconds.
-func collectMetrics(ctx context.Context, port, dirPath string) {
+func collectMetrics(ctx context.Context, port, dirPath string, scrapeInterval int) {
+	genNumber := 1
 	metricEndpoint := "http://localhost:" + port + "/metrics"
-	outputPath := dirPath + "/metrics.prom"
-	ticker := time.NewTicker(10 * time.Second)
+	ticker := time.NewTicker(time.Duration(scrapeInterval) * time.Second)
 
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
 			newCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
-			scrapeMetrics(newCtx, metricEndpoint, outputPath)
+			scrapeMetrics(newCtx, metricEndpoint, dirPath, genNumber)
+			// If x is the current generation of the metrics snapshot/file, we are leaving some buffer where the outdated (x-1)
+			// file exists so that the reader has time to finish reading the complete data.
+			// The -2 means we are deleting file generations up to x - 2.
```

Comment:
Michelle suggested that if the writer deletes a file while the reader holds a file descriptor, the kernel should not delete the file until the file descriptor is released by all processes. 

We can test if things work fine by deleting all outdated files, and only keeping the most recent.

Reply:
Will keep the current approach as the following scenario fails if removed:

Scenario:
1. Reader: reads directory and finds latest file is metrics_x.prom
2. Writer : creates metrics_y.prom & deletes metrics_x.prom
3. Reader: attempts to open metrics_x.prom and fails